ndless-sdk/toolchain/build_toolchain.sh

```diff
@@ -39,38 +39,43 @@ mkdir -p "$PREFIX"
-if ! gcc -lmpc test.c -o test; then error=1; echo 'MPC (mpc-devel/libmpc-dev) dependency seems to be missing!'; fi
-if ! gcc -lz test.c -o test; then error=1; echo 'zlib (zlib-devel/zlib1g-dev) dependency seems to be missing!'; fi
+if ! gcc -lmpc  test.c -o test; then error=1; echo 'MPC (mpc-devel/libmpc-dev) dependency seems to be missing!'; fi
+if ! gcc -lz    test.c -o test; then error=1; echo 'zlib (zlib-devel/zlib1g-dev) dependency seems to be missing!'; fi
 #TODO: This test fails on openSUSE
 #if ! gcc -ltermcap test.c -o test; then error=1; echo 'termcap (termcap/libtinfo-dev) dependency seems to be missing!'; fi
 if ! gcc -lpython2.7 test.c -o test; then error=1; echo 'libpython2.7 (python-devel/python2.7-dev) dependency seems to be missing!'; fi
 rm -f test test.c
-if ! makeinfo -h > /dev/null; then error=1; echo 'makeinfo (texinfo) dependency seems to be missing!'; fi
+if ! makeinfo -h > /dev/null;   then error=1; echo 'makeinfo (texinfo) dependency seems to be missing!'; fi
 [ $error -eq 1 ] && exit 1
 
 mkdir -p build download
 
 if [ ! -f .downloaded ]; then
 	wget -c http://ftp.gnu.org/gnu/binutils/$BINUTILS.tar.bz2 -O download/$BINUTILS.tar.bz2 && tar xvjf download/$BINUTILS.tar.bz2 && \
-	wget -c ftp://ftp.gnu.org/gnu/gcc/$GCC/$GCC.tar.bz2 -O download/$GCC.tar.bz2 && tar xvjf download/$GCC.tar.bz2 && \
-	wget -c ftp://ftp.gnu.org/gnu/gdb/$GDB.tar.xz -O download/$GDB.tar.xz && tar xvJf download/$GDB.tar.xz && \
-	wget -c ftp://sourceware.org/pub/newlib/$NEWLIB.tar.gz -O download/$NEWLIB.tar.gz && tar xvzf download/$NEWLIB.tar.gz && \
+	wget -c ftp://ftp.gnu.org/gnu/gcc/$GCC/$GCC.tar.bz2       -O download/$GCC.tar.bz2      && tar xvjf download/$GCC.tar.bz2 && \
+	wget -c ftp://ftp.gnu.org/gnu/gdb/$GDB.tar.xz             -O download/$GDB.tar.xz       && tar xvJf download/$GDB.tar.xz && \
+	wget -c ftp://sourceware.org/pub/newlib/$NEWLIB.tar.gz    -O download/$NEWLIB.tar.gz    && tar xvzf download/$NEWLIB.tar.gz && \
 	touch .downloaded
 	if [ $? -ne 0 ]; then
 		echo "Download failed!"
 		exit 1
 	fi
 fi
 
+EXTRAGCCOPTS=''
+if [ ! $(gcc -v 2>&1 | grep clang > /dev/null) ]; then
+    EXTRAGCCOPTS='CXXFLAGS="-fbracket-depth=512"'
+fi
+
 # Section 1: GNU Binutils.
 echo "Building Binutils..."
 [ -f .built_binutils ] || (cd build && rm -rf * && ../$BINUTILS/configure $OPTIONS_BINUTILS && make $PARALLEL all && make install && cd .. && rm -rf build/* && touch .built_binutils) || exit 1;
 
 # Section 2: GCC, step 1.
 echo "Building GCC (step 1)..."
-[ -f .built_gcc_step1 ] || (cd build && rm -rf * && ../$GCC/configure $OPTIONS_GCC && make $PARALLEL all-gcc && make install-gcc && cd .. && rm -rf build/* && touch .built_gcc_step1) || exit 1;
+[ -f .built_gcc_step1 ] || (cd build && rm -rf * && ../$GCC/configure $OPTIONS_GCC && make $PARALLEL $EXTRAGCCOPTS all-gcc && make install-gcc && cd .. && rm -rf build/* && touch .built_gcc_step1) || exit 1;
```

Comment:
Please move to configure as it's also done on https://github.com/busterb/musl-cross-make/blob/fcf2604a5a1c91f2fd590af83c2dc691a6ee49a2/litecross/Makefile